lib/pbench/server/api/resources/users_api.py

```diff
@@ -0,0 +1,533 @@
+            abort(500, message="INTERNAL ERROR")
+
+        # TODO: Check if the user has the right privileges
+        if verified:
+            return_data = user.get_json()
+            response_object = {
+                "status": "success",
+                "data": return_data,
+            }
+            return make_response(jsonify(response_object), 200)
+        elif user.is_admin():
+            try:
+                target_user = User.query(username=username)
+                return_data = target_user.get_json()
+                response_object = {
+                    "status": "success",
+                    "data": return_data,
+                }
+                return make_response(jsonify(response_object), 200)
+            except Exception:
+                self.logger.exception(
+                    "Exception occurred while querying the user. Username: {}", username
+                )
+                abort(500, message="INTERNAL ERROR")
+        else:
+            self.logger.warning(
+                "Username retrieved from the auth token {} is different from the username provided",
+                self.auth.token_auth.current_user().id,
+            )
+            abort(403, message="Authentication token does not belong to the user")
+
+    @Auth.token_auth.login_required(f=Auth().verify_auth())
+    def put(self, username):
+        """
+        PUT request for updating user data.
+        This requires a Pbench auth token in the header field
+
+        This requires a JSON data with required user registration fields that needs an update
+        Example Json:
+        {
+            "first_name": "new_name",
+            "password": "password",
+            ...
+        }
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            Authorization:  Bearer Pbench_auth_token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "message": "Successfully registered."/"failure message",
+                    "data": {
+                        "username": <username>,
+                        "first_name": <firstName>,
+                        "last_name": <lastName>,
+                        "registered_on": registered_on,
+                    }
+                }
+        """
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: {}", request.url)
+            abort(400, message="Invalid json object in request")
+
+        try:
+            user, verified = self.auth.verify_user(username)
+        except Exception:
+            self.logger.exception("Exception occurred while verifying the user")
+            abort(500, message="INTERNAL ERROR")
+
+        # TODO: Check if the user has the right privileges
+        if verified:
```

Comment:
Let's file an issue to address this comment later:

The `else` of the condition clause of this condition should be checked first so that we can avoid indenting this entirely block.

Reply:
fixed